### PR TITLE
chore(releasing): Prepare v0.40.0 release

### PR DESCRIPTION
**Pull request by jszwedko**
Checklist: https://github.com/vectordotdev/vector/issues/20923

### CODE REVIEW

**Review by pront — APPROVED** [review]
LGTM. Is there a preview for the upgrade guide? I wonder if it renders nicely.

**Comment by jszwedko** [comment]
> LGTM. Is there a preview for the upgrade guide? I wonder if it renders nicely.

You can find it rendered here: https://jszwedko-website-prepare-v0-40-0.d1a7j77663uxsc.amplifyapp.com/highlights/2024-07-29-0-40-0-upgrade-guide/

File: website/content/en/highlights/2024-07-29-0-40-0-upgrade-guide.md [hunk 1]
```diff
@@ -0,0 +1,166 @@
...
+  type: breaking change
+---
+
+Vector's 0.40.0 release includes three **breaking changes**:
+
+1. [GELF codec defaults to null-delimited messages for stream-based sources](#gelf)
+1. [Reduce transforms now properly aggregate nested fields](#reduce-aggregate)
+1. [Vector no longer supports CentOS 7](#centos7)
+
+We cover them below to help you upgrade quickly:
```
Comment (brett0000FF) [inline website/content/en/highlights/2024-07-29-0-40-0-upgrade-guide.md:18]:
```suggestion
The changes are covered below to help you upgrade quickly:
```

**Review by brett0000FF — APPROVED** [review]
One small comment. Thanks!

**Comment by jszwedko** [comment]
Merged locally.

**Pull request closed without merging**